lighthouse-core/fraggle-rock/user-flow.js

```diff
@@ -176,6 +166,69 @@ class UserFlow {
     const flowResult = await this.createFlowResult();
     return generateFlowReportHtml(flowResult);
   }
+
+  /**
+   * @return {LH.UserFlow.FlowArtifacts}
+   */
+  createArtifactsJson() {
+    /** @type {LH.UserFlow.GatherStep[]} */
+    const stepArtifacts = this._gatherSteps.map(step => ({
+      name: step.name,
+      artifacts: step.artifacts,
+      config: step.config,
+      configContext: step.configContext,
+    }));
+
+    return {
+      gatherSteps: stepArtifacts,
+      name: this.name,
+    };
+  }
+}
+
+/**
+ * @param {Array<LH.UserFlow.GatherStep | LH.UserFlow.ActiveGatherStep>} gatherSteps
+ * @param {string|undefined} name
+ * @param {LH.Config.Json|undefined} flowConfig Must be the same as the config provided when the flow started.
+ */
+async function auditGatherSteps(gatherSteps, name, flowConfig) {
+  if (!gatherSteps.length) {
+    throw new Error('Need at least one step before getting the result');
+  }
+
+  /** @type {LH.FlowResult['steps']} */
+  const steps = [];
+  for (const gatherStep of gatherSteps) {
+    const {artifacts, name, configContext} = gatherStep;
+
+    let runnerOptions;
+    if ('runnerOptions' in gatherStep) {
```

Comment:
```suggestion
    if (gatherStep.runnerOptions) {
```

+ just move `runnerOptions` into the type and make it optional? could maybe rename it to `activeRunnerOptions`.

Reply:
Ya that's simpler, I like